hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/XceiverClientManager.java

```diff
@@ -297,6 +293,15 @@ private String getPipelineCacheKey(Pipeline pipeline,
             e.getMessage());
       }
     }
+    // Append user short name to key to prevent a different user
+    // from using same instance of xceiverClient.
+    try {
+      key = isSecurityEnabled ?
+          key + UserGroupInformation.getCurrentUser().getShortUserName() : key;
+    } catch (IOException e) {
+      LOG.error("Failed to get current user to create pipeline cache key:" +
+          e.getMessage());
+    }
```

Comment:
We can simplify this (and avoid the `try-catch` if security is not enabled).

```suggestion
    if (isSecurityEnabled) {
      try {
        key += UserGroupInformation.getCurrentUser().getShortUserName();
      } catch (IOException e) {
        LOG.error("Failed to get current user to create pipeline cache key: " +
            e.getMessage());
      }
    }
```

Reply:
@adoroszlai , thank you for the review. It's addressed.